pkg/ip/cidr.go

```diff
@@ -71,6 +71,21 @@ func PrefixToIPNet(prefix netip.Prefix) *net.IPNet {
 	}
 }
 
+// IPNetToPrefix is a convenience helper for migrating from the older 'net'
+// standard library types to the newer 'netip' types. Use this to plug the
+// new types in newer code into older types in older code during the migration.
+func IPNetToPrefix(prefix *net.IPNet) netip.Prefix {
```

Comment:
I think this equivalent to [`netipx.FromStdIPNet`](https://pkg.go.dev/go4.org/netipx#FromStdIPNet) (sans the additional `bool` return). Given that we already vendor that library I wonder whether we could use that conversion function instead of adding our own conversion helper?

Also ref. https://github.com/cilium/cilium/issues/24246#issuecomment-1547961878

Reply:
Ah, I forgot about this. Sending a PR to revert and fixup the use of `net` pkg. Thanks for catching this!